README
Amarok Full Screen GUI
		 
You need Amarok running. For any information about the player itseft, please visit amarok.kde.org
For quit press Esc or Alt+F4

INSTALL:
	
	Executable is installed into /usr/bin, 
	default theme + its images into /usr/share/amarokFS, 

THEMES:
	The theme can be chosen by the button in the upper left corner (which hides automaticaly).
	If no theme is selected (or the selected theme is not readable) the default theme is used (/usr/share/amarokFS/theme.xml).

	If you want to write your own skins please read the instructions in theme-howto.txt.

CHANGELOG:
	0.1: first release
	
	0.2:
		ESC for Exit
		fancy reflection
	
	0.2.1:
		made Makefile more compatible with other distros
	
	0.3:
		new methods for searching cover images
		Makefile now supports "make install"
	
	0.4:
		initial THEME SUPPORT!!
		mouse wheel for volume control
		updated Makefile (should work on gentoo now)

	0.4.1:
		gui dialog to choose the theme
		new themable elements

	0.4.2:
		disabling screen saver (optional)
		
	0.5:
		many fixes and updates by Martin Kossler - thanks a lot!
		better progessbar
		new element - labelPixmap
		new element - nextPlaying
		context menu instead of settings button
		mouse hiding
		added amarok script 
				
Copyright (C) 2007 Jan Zarnikov <jan.zarnikov@gmail.com>
		Martin Kossler <martin.kossler@students.uni-marburg.de> 

	This file is free software; you can redistribute it and/or
	modify it under the terms of the GNU Library General Public
	License as published by the Free Software Foundation; either
	version 2 of the License, or (at your option) any later version.

	This library is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
	Library General Public License for more details.

	You should have received a copy of the GNU Library General Public License
	along with this library; see the file COPYING.LIB.  If not, write to
	the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
	Boston, MA 02110-1301, USA.

		


debian/changelog
amarokfs (0.4.2-9) unstable; urgency=low

  * mouse cursor will not hide when contextmenu or dialog shows up

 -- Martin Koßler <martin.kossler@students.uni-marburg.de>  Sun, 18 Mar 2007 23:23:06 +0100

amarokfs (0.4.2-8) unstable; urgency=low

  * added application icon
  * added about information
  * added command line options:
    + show config dialog
    + kiosk-mode: disable settings entry in context menu
  * config dialog is accessible from Amarok script management

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Thu, 15 Mar 2007 17:35:19 +0100

amarokfs (0.4.2-7) unstable; urgency=low

  * mouse cursor will hide after five seconds
  * added contextmenu instead of settings button containing reload function
    for nextPlaying element

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Mon, 12 Mar 2007 01:38:20 +0100

amarokfs (0.4.2-6) unstable; urgency=low

  * Added List of next Songs in playlist. This is to be activated in the theme
    through the new nextPlaying element
  * Fixed Amarok-Script so that it will show multiple menu options only if
    more than one X-Screen exists

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Fri,  2 Mar 2007 19:23:38 +0100

amarokfs (0.4.2-5) unstable; urgency=low

  * progressbar now changes every second
  * theme fix: pause button now triggers playPause

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Thu,  1 Mar 2007 17:50:09 +0100

amarokfs (0.4.2-4) unstable; urgency=low

  * Config Dialog Layout patched with code from fedora cvs
  * fixed typos in theme-howto.txt
  * Updated paths in Readme 

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Wed, 28 Feb 2007 17:27:32 +0100

amarokfs (0.4.2-3) unstable; urgency=low

  * better gradient.png (svg with radial gradient converted to png)
  * If Cover is less than 300px high it will be aligned at the bottom to avoid
    the gap between cover and reflection

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Wed, 21 Feb 2007 18:04:10 +0100

amarokfs (0.4.2-2) unstable; urgency=low

  * the patch for ignoring stupid dcop-results only worked for trackLabel, now
    for instantLabel too.
  * positioning fix on standard theme
  * better multi-screen-hack (but not right optimal)

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Thu, 15 Feb 2007 19:28:42 +0100

amarokfs (0.4.2-1) unstable; urgency=low

  * Version 0.4.2 from kde-apps.de by Jan Zarnikov <jan.zarnikov@gmail.com>
  * patched to use common ubuntu-paths (/usr/ instead of /usr/local/)
  * Added and modfied patch that allows to disable the OSD while running
    amarokFS (original posted to the comments at kde-apps.de by Matthias
    Fehring) and to disable the hint shown at startup.
  * Patch to the cover-extractor - now the correct cover is chosen independent
    from the size of the cover-preview
  * Fixed behavior on dual-screen setup: The setting button won't show up any
    more when mouse is on second screen.
  * uses modified isimplydark-theme as standard
  * Added amarok-script so that we are able to access amarokFS from
    within amarok. So the desktopmenu entry can also get removed.
  * Added xml-tag for theme-file which allows to supress stupid dcop-results.
    Instead a defaultText can be defined.

 -- Martin Kossler <martin.kossler@students.uni-marburg.de>  Wed, 14 Feb 2007 19:15:31 +0100


main.cpp
/*
	This file is part of the Amarok Full Screen GUI
	Copyright (C) 2007 Jan Zarnikov (jan.zarnikov@gmail.com)

	This file is free software; you can redistribute it and/or
	modify it under the terms of the GNU Library General Public
	License as published by the Free Software Foundation; either
	version 2 of the License, or (at your option) any later version.

	This library is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
	Library General Public License for more details.

	You should have received a copy of the GNU Library General Public License
	along with this library; see the file COPYING.LIB.  If not, write to
	the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
	Boston, MA 02110-1301, USA.
*/

#include <qapplication.h>
#include <kapplication.h>
#include <kaboutdata.h>
#include <kcmdlineargs.h>
#include <kaboutapplication.h>
#include <qimage.h>
#include <vector>
#include "amfs-xml.h"
#include "amarokfs_config.h"

using namespace std;

int main( int argc, char ** argv )
{
	KAboutData* about = new KAboutData("amarokFS", "Amarok Full Screen", "0.4.2",
		"A full screen front end for Amarok", KAboutData::License_GPL_V2,
		"© 2007, Jan Zarnikov", 0,
		"http://www.kde-apps.org/content/show.php?content=52641");
	about->addAuthor("Jan Zarnikov", 0, "jan.zarnikov@gmail.com");
	about->addAuthor("Martin Koßler", 0, "martin.kossler@students.uni-marburg.de");
	KCmdLineArgs::init(argc, argv, about);
	KCmdLineOptions options[] =
	{
	    { "settings", "show config dialog", 0 },
	    { "kiosk", "disable Settings entry in contextmenu", 0 },
	    KCmdLineLastOption
	};
	KCmdLineArgs::addCmdLineOptions(options);
	KCmdLineArgs* args = KCmdLineArgs::parsedArgs();

	KApplication a;
	QWidget* x;

	if(args->isSet("settings")) {
		x = new AmarokFS_config();
		x->show();
	} else {
		x = new AmfsXML();
		if(args->isSet("kiosk"))
			((AmfsXML*)x)->setupAct->setEnabled(FALSE);
		((AmfsXML*)x)->aboutWindow = new KAboutApplication(about, x, "about", FALSE);
		x->showFullScreen();
	}

	x->setIcon(QImage("/usr/share/icons/amarokFS.png"));
	a.setMainWidget(x);
	return a.exec();
}
	
